kytos/core/api_server.py

```diff
@@ -130,3 +134,65 @@ def shutdown_api(self):
     def web_ui(self):
         """Method used to serve the index.html page for the admin-ui."""
         return send_from_directory(self.flask_dir, 'index.html')
+
+    # BEGIN decorator methods
+
+    @staticmethod
+    def decorator(rule, **options):
+        """Decorator for REST endpoints using Flask.
+
+        Parameter documentation is the same as Flask's @app.route. As we don't
+        have the NApp instance now, we store the parameters in a method
+        attribute in order to add the route later, after APIServer is the NApp
+        object is passed as a parameter of :meth:`start_rest_api`.
+        """
+        def store_route_params(function):
+            """Store ``Flask`` ``@route`` parameters in a method attribute."""
+            function.route_params = rule, options
+            return function
+        return store_route_params
+
+    def start_rest_api(self, napp):
+        """Start ``napp`` REST endpoints."""
```

Comment:
Let's try to make APIServer NAPP agnostic/independent. `s/napp/instance/g`  for instance.

For the APIServer perspective doesn't matter if is a napp or other object, right ?